implementations/prometheus/queue.go

```
@@ -109,11 +109,15 @@ func NewQueue(name string, cc types.ConnectionConfig, directory string, maxSigna
 	return q, nil
 }
 
-func (q *queue) Start(ctx context.Context) {
+func (q *queue) Start(ctx context.Context) error {
 	q.network.Start(ctx)
 	q.queue.Start(ctx)
-	q.serializer.Start(ctx)
+	err := q.serializer.Start(ctx)
+	if err != nil {
+		return err
```

should the network and the queue be stopped here? Also this function can now return an error but the error is not checked in the calling code

---

Nothing that I noticed while testing as far as memory. There is likely a small bump but in the great scheme of things its only used for a very short window.

---

Good call on stopping, adding that in.